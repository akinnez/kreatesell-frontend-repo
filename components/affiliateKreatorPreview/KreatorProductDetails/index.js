import {HiOutlineExternalLink} from 'react-icons/hi';
import formatNumber from 'utils/formatNumber';
import productPriceFn from 'utils/productPriceFn';
import styles from './index.module.scss';

const KreatorProductDetails = ({stock, sold, productName, checkoutDetails}) => {
	const priceDetails = productPriceFn(checkoutDetails);

	return (
		<div className={styles.product_details}>
			<div className={styles.product__stats}>
				{stock && (
					<span
						className={
							stock === 'In Stock'
								? styles['product--in-stock']
								: stock === 'Out of Stock'
								? styles['product--out-of-stock']
								: styles.product__stock
						}
					>
						{stock}
					</span>
				)}
				{stock && sold > 0 && (
					<>
						&nbsp;&nbsp;
						<span className={styles.product__stats__divider}>
							|
						</span>
						&nbsp;&nbsp;
					</>
				)}
				{sold > 0 && (
					<span className={styles.product__sold}>
						{formatNumber(sold)} Sold
					</span>
				)}
			</div>
			<div className={styles.product__name}>{productName}</div>
			<div className={styles.product__price}>
				<span>
					{priceDetails ? (
						<>
							{priceDetails.currency}{' '}
							{formatNumber(priceDetails.price)}
						</>
					) : (
						'0.00'
					)}
				</span>
				<HiOutlineExternalLink />
			</div>
		</div>
	);
};

export default KreatorProductDetails;
