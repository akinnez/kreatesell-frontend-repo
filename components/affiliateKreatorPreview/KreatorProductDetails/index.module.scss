@use "public/css/colors";
@use "public/css/mixins";

.product_details {
  background-color: colors.$base-white;
  padding: 15px 24px 30px;
}

.product__stats {
  min-height: 30px;
  font-size: 16px;
  margin-bottom: 8px;
}

.product--in-stock {
  color: colors.$base-green-100;
}

.product--out-of-stock {
  color: colors.$error-main;
}

.product__stats__divider {
  color: colors.$white-300;
}

.product__stock,
.product__sold {
  color: colors.$gray-200;
}

.product__name {
  font-weight: 700;
  font-size: 18px;
  color: colors.$black-100;
  margin-bottom: 10px;
}

.product__price {
  @include mixins.flex(row, space-between, center);

  span {
    font-weight: 500;
    font-size: 20px;
    color: colors.$gray-200;
  }

  svg {
    color: colors.$gray-700;
    font-size: 30px;
  }
}
