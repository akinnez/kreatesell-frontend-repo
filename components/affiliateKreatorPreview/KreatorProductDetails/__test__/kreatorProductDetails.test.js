import KreatorProductDetails from '..';
import renderer from 'react-test-renderer';

const mockProps = {
	stock: true,
	productName: 'mock Product',
};

describe('KreatorProductDetails: ', () => {
	it('renders a snapshot of <KreatorProductDetails /> with no props', () => {
		const el = renderer.create(<KreatorProductDetails />).toJSON();
		expect(el).toMatchSnapshot();
	});
	it('renders a snapshot of <KreatorProductDetails /> with mock props and sold at 0', () => {
		const el = renderer
			.create(<KreatorProductDetails {...mockProps} sold={0} />)
			.toJSON();
		expect(el).toMatchSnapshot();
	});
	it('renders a snapshot of <KreatorProductDetails /> with mock props and sold > 0', () => {
		const el = renderer
			.create(<KreatorProductDetails {...mockProps} sold={10} />)
			.toJSON();
		expect(el).toMatchSnapshot();
	});
});
