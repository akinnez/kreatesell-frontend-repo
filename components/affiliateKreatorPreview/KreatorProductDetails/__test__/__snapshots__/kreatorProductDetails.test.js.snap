// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KreatorProductDetails:  renders a snapshot of <KreatorProductDetails /> with mock props and sold > 0 1`] = `
<div
  className="product_details"
>
  <div
    className="product__stats"
  >
    <span
      className="product__stock"
    />
      
    <span
      className="product__stats__divider"
    >
      |
    </span>
      
    <span
      className="product__sold"
    >
      10
       Sold
    </span>
  </div>
  <div
    className="product__name"
  >
    mock Product
  </div>
  <div
    className="product__price"
  >
    <span>
      0.00
    </span>
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10 6H6a2 2 0 00-2 2v10a2 2 0 002 2h10a2 2 0 002-2v-4M14 4h6m0 0v6m0-6L10 14"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
      />
    </svg>
  </div>
</div>
`;

exports[`KreatorProductDetails:  renders a snapshot of <KreatorProductDetails /> with mock props and sold at 0 1`] = `
<div
  className="product_details"
>
  <div
    className="product__stats"
  >
    <span
      className="product__stock"
    />
  </div>
  <div
    className="product__name"
  >
    mock Product
  </div>
  <div
    className="product__price"
  >
    <span>
      0.00
    </span>
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10 6H6a2 2 0 00-2 2v10a2 2 0 002 2h10a2 2 0 002-2v-4M14 4h6m0 0v6m0-6L10 14"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
      />
    </svg>
  </div>
</div>
`;

exports[`KreatorProductDetails:  renders a snapshot of <KreatorProductDetails /> with no props 1`] = `
<div
  className="product_details"
>
  <div
    className="product__stats"
  />
  <div
    className="product__name"
  />
  <div
    className="product__price"
  >
    <span>
      0.00
    </span>
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10 6H6a2 2 0 00-2 2v10a2 2 0 002 2h10a2 2 0 002-2v-4M14 4h6m0 0v6m0-6L10 14"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
      />
    </svg>
  </div>
</div>
`;
